test/devicecontroller/DefaultDeviceController.test.ts

```diff
@@ -2398,28 +2398,102 @@ describe('DefaultDeviceController', () => {
       element = videoElementFactory.create();
     });
 
-    it('connects the video stream to the video element', async () => {
+    it('connects the transform video stream to the video element if there is a transform device stream', async () => {
       const spy1 = sinon.spy(DefaultVideoTile, 'disconnectVideoStreamFromVideoElement');
       const spy2 = sinon.spy(DefaultVideoTile, 'connectVideoStreamToVideoElement');
       const gUM = sinon.spy(navigator.mediaDevices, 'getUserMedia');
+      domMockBehavior.createElementCaptureStream = new MediaStream();
+      domMockBehavior.mediaStreamTrackSettings.deviceId = stringDeviceId;
+      domMockBehavior.createElementCaptureStream.addTrack(new MediaStreamTrack());
+      // Connect the original video stream
       await deviceController.chooseVideoInputDevice(stringDeviceId);
       deviceController.startVideoPreviewForVideoInput(element);
-      expect(spy1.calledOnce).to.be.true;
+      const processor = new NoOpVideoFrameProcessor();
+      const device = new DefaultVideoTransformDevice(logger, stringDeviceId, [processor]);
+      // Connect the transform video stream
+      await deviceController.chooseVideoInputDevice(device);
+      deviceController.startVideoPreviewForVideoInput(element);
+
+      expect(device.getInnerDevice() === stringDeviceId);
+      expect(element.srcObject === (await gUM.returnValues.pop()));
+      expect(spy1.notCalled).to.be.true;
+      expect(spy2.calledTwice).to.be.true;
+      expect(gUM.calledOnce).to.be.true;
+      spy1.restore();
+      spy2.restore();
+      gUM.restore();
+    });
+
+    it('connects the active device video stream to the video element if there is no transform device stream', async () => {
+      const spy1 = sinon.spy(DefaultVideoTile, 'disconnectVideoStreamFromVideoElement');
+      const spy2 = sinon.spy(DefaultVideoTile, 'connectVideoStreamToVideoElement');
+      const gUM = sinon.spy(navigator.mediaDevices, 'getUserMedia');
+      await deviceController.chooseVideoInputDevice(stringDeviceId);
+
+      deviceController.startVideoPreviewForVideoInput(element);
+
+      expect(gUM.returnValues).to.have.lengthOf(1);
+      expect(element.srcObject === (await gUM.returnValues.pop()));
+      expect(spy1.notCalled).to.be.true;
       expect(spy2.calledOnce).to.be.true;
       expect(gUM.calledOnce).to.be.true;
       spy1.restore();
       spy2.restore();
       gUM.restore();
     });
 
-    it('does not disconnect or connect the video stream for preview if no active video exists', () => {
+    it('reuses the original video stream to create a transform device stream', async () => {
```

Comment:
What is the different between this one and the one above `connects the transform video stream to the video element if there is a transform device stream` ?

Reply:
There is a slight difference between the two. Reuse one is not expecting the following two expect statements. My goal with `connects the transform video stream to the video element if there is a transform device stream` is that the stream on the canvas element is the transform stream and not the original stream.

```expect(device.getInnerDevice() === stringDeviceId);
expect(element.srcObject === (await gUM.returnValues.pop()));```